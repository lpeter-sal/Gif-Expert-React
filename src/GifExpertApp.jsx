import { useState } from 'react';
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState([]);

    const onAddCategory = (newCategory) => {

        if(categories.includes(newCategory)) return;

        // setCategories(categories.concat('Valorant'));
        // setCategories([...categories, 'Valorant']);
        setCategories([newCategory, ...categories]);
    }


    const resetCategory = () => {
        setCategories([]);
    }

  return (
    <>

        <h1>GifExpertApp</h1>


        <AddCategory 
            onNewCategory={ (event) => onAddCategory(event)}
        />

        <button className='btn primary' onClick={resetCategory}>Reset</button>

        { 
            categories.map( (category) => (
                <GifGrid key={category} 
                category={category} />
            ))
        }



    </>
  )
}
