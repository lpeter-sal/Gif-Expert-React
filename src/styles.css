* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(234, 234, 234);
  }
  
  body {
    padding: 60px;
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }


  .btn {
    border: none; 
    border-radius: 5px; 
    color: white; 
    cursor: pointer; 
    margin-top: 5px;
    margin-right: 5px;
    padding: 14px 28px; 
  }
  
  .primary {background-color: #007bff;} 
  .primary:hover {background: #0b7dda;}
  
  .secondary {background-color: #e7e7e7; color: black;} 
  .secondary:hover {background: #ddd;}
  
  .success {background-color: #04AA6D;} 
  .success:hover {background-color: #46a049;}
  
  .warning {background-color: #ff9800;} 
  .warning:hover {background: #e68a00;}
  
  .danger {background-color: #f44336;} 
  .danger:hover {background: #da190b;}
  