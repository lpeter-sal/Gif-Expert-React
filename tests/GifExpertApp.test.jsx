import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"


describe('Pruebas en <GifExpertApp />', () => { 
    
    test('Debe hacer match con el snapshot', () => {
        
        const { container } = render( <GifExpertApp /> );
        expect(container).toMatchSnapshot();

    });

    test('Debe añadir una categoría', () => {
 
        render( <GifExpertApp />);
 
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
       
        fireEvent.input( input, { target: { value:  'Iroman'} } );
        fireEvent.submit( form );
        
        expect( screen.getByText('Iroman') ).toBeTruthy();
    });


});