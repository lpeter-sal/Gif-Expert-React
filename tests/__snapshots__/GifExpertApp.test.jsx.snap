// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <button
    class="btn primary"
  >
    Reset
  </button>
</div>
`;
